server/src/main/java/org/elasticsearch/ExceptionsHelper.java

```diff
@@ -175,10 +175,26 @@ public static String formatStackTrace(final StackTraceElement[] stackTrace) {
         return first;
     }
 
+    private static final List<Class<? extends IOException>> CORRUPTION_EXCEPTIONS =
+        List.of(CorruptIndexException.class, IndexFormatTooOldException.class, IndexFormatTooNewException.class);
+
     public static IOException unwrapCorruption(Throwable t) {
```

Comment:
can you add javadocs to mention that this checks both causes and suppressed exceptions?

Reply:
and maybe javadocs to `unwrap` too to make it clear it **doesn't** look at suppressed exceptions?

Reply:
Sounds good, added javadoc to both :)